pkgs/applications/virtualization/singularity/generic.nix

```diff
@@ -0,0 +1,199 @@
+      if [ -f "$subPath" ]; then
+        echo "Patching [Dd]efaultPath in \"$subPath\""
+        sed -i 's|^\([ \t]*[Dd]efaultPath[ \t]*[:]\?=[ \t]*\)"[^"]*"|\1"${lib.makeBinPath [ coreutils squashfsTools bash ]}"|' "$subPath"
+      fi
+    done
+
+    # Try to make use of the existing configureFlags and configureFlagsArray
+    # Code borrowed from pkgs/stdenv/generic/setup.sh configurePhase()
+
+    # set to empty if unset
+    : ''${configureScript=}
+    : ''${configureFlags=}
+
+    if [[ -z "''${dontAddPrefix:-}" && -n "$prefix" ]]; then
+        configureFlags="''${prefixKey:---prefix=}$prefix $configureFlags"
+    fi
+
+    # Old bash empty array hack
+    # shellcheck disable=SC2086
+    flagsArray=( $configureFlags "''${configureFlagsArray[@]}" )
+    echoCmd 'configure flags' "''${flagsArray[@]}"
+    # shellcheck disable=SC2086
+    $configureScript -V ${version} "''${flagsArray[@]}"
+    unset flagsArray
+
+    # End of the code from pkgs/stdenv/generic/setup.sh configurPhase
+
+    # Don't install SUID binaries
+    sed -i 's/-m 4755/-m 755/g' builddir/Makefile
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make -C builddir
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    make -C builddir install LOCALSTATEDIR=$out/var
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    # Explicitly configure paths in the config file
+    sed -i 's|^# mksquashfs path =.*$|mksquashfs path = ${lib.makeBinPath [squashfsTools]}/mksquashfs|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# unsquashfs path =.*$|unsquashfs path = ${lib.makeBinPath [squashfsTools]}/unsquashfs|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# cryptsetup path =.*$|cryptsetup path = ${lib.makeBinPath [cryptsetup]}/cryptsetup|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# go path =.*$|go path = ${lib.makeBinPath [go]}/go|' $out/etc/${projectName}/${projectName}.conf
+    ${lib.optionalString enableNvidiaContainerCli ''
+      sed -i 's|^# nvidia-container-cli path =.*$|nvidia-container-cli path = ${lib.makeBinPath [nvidia-docker]}/nvidia-container-cli|' $out/etc/${projectName}/${projectName}.conf
+    ''}
+
+    ${lib.optionalString (removeCompat && (projectName != "singularity")) ''
+      while IFS= read -r -d $'\0' filePath; do
+        if [ -f "$(realpath "$filePath")" ]; then
+          fileDir="$(dirname "$filePath")"
+          fileName="$(basename "$filePath")"
+          fileNameReplaced="''${fileName/singularity/${projectName}}"
+          if [ -e "$fileDir/$fileNameReplaced" ] && [ "$(realpath "$filePath")" == "$(realpath "$fileDir/$fileNameReplaced")" ]; then
+            unlink "$filePath"
+          fi
+        fi
+      done < <(find "$out" -mindepth 1 -type l -name "*singularity*" -print0)
+      for filePath in "$out"/share/*-completion/completions/singularity; do
+        if [ -e "$filePath" ]; then
+          rm "$filePath"
+        fi
+      done
```

Comment:
As I understand it we want to remove a symlink here. Why do we so much code for that? Isn't a ``rm $out/some/where`` enough?

Reply:
I have modified the misleading comment. It's actually not a single symlink, but a bunch of them.
The removal should be more specific anyway.